sys/include/net/dns.h

```diff
@@ -43,6 +53,51 @@ extern "C" {
 #define RR_RDLENGTH_LENGTH  (2U)
 /** @} */
 
+/**
+ * @brief Get IP address for DNS name
+ *
+ * This function will synchronously try to resolve a DNS A or AAAA record by contacting
+ * the DNS server specified in the global variable @ref sock_dns_server.
+ *
+ * By supplying AF_INET, AF_INET6 or AF_UNSPEC in @p family requesting of A
+ * records (IPv4), AAAA records (IPv6) or both can be selected.
+ *
+ * This function will return the first DNS record it receives. IF both A and
+ * AAAA are requested, AAAA will be preferred.
+ *
+ * @note @p addr_out needs to provide space for any possible result!
+ *       (4byte when family==AF_INET, 16byte otherwise)
+ *
+ * @param[in]   domain_name     DNS name to resolve into address
+ * @param[out]  addr_out        buffer to write result into
+ * @param[in]   family          Either AF_INET, AF_INET6 or AF_UNSPEC
+ *
+ * @return      the size of the resolved address on success
+ * @return      < 0 otherwise
+ */
+static inline int dns_query(const char *domain_name, void *addr_out, int family)
+{
+    int res = -ENOTSUP;
+
+#if defined(MODULE_GCOAP_DNS)
+    if (res < 0) {
+        res = gcoap_dns_query(domain_name, addr_out, family);
+    }
+#endif
+#if defined(MODULE_SOCK_DODTLS)
+    if (res < 0) {
+        res = sock_dodtls_query(domain_name, addr_out, family);
+    }
+#endif
+#if defined(MODULE_SOCK_DNS) || defined(MODULE_SOCK_DNS_MOCK)
+    if (res < 0) {
+        res = sock_dns_query(domain_name, addr_out, family);
+    }
+#endif
```

Comment:
```suggestion
    if ((IS_USED(MODULE_SOCK_DNS) || IS_USED(MODULE_SOCK_DNS_MOCK)) && res < 0) {
        res = sock_dns_query(domain_name, addr_out, family);
    }
```

Reply:
I really don’t see what’s the advantage of `IS_USED` outside C code